### ISSUE TITLE
Improve argument type coercing by inspecting the ABI

### ISSUE THREAD

**Issue by 0x6431346e** [aragon/aragon-cli#320]
When trying to create a new dao from a `multisig-kit`, we need to call `newTokenAndInstance` with 
```
       string name,
       string symbol,
       address[] signers,
       uint256 neededSignatures
```
Note that signers here needs to be an array, so this won't work:

```
> aragon dao new multisig-kit --fn newTokenAndInstance --fn-args Aragon ANT 0xb4124cEB3451635DAcedd11767f004d8a28c6eE7 1

expected array value (arg="signers", coderType="array", value="0xb4124cEB3451635DAcedd11767f004d8a28c6eE7")
```

This won't work either:
```
> aragon dao new multisig-kit --fn newTokenAndInstance --fn-args Aragon ANT '["0xb4124cEB3451635DAcedd11767f004d8a28c6eE7"]' 1
```
because signers is still a string:
```
fnArgs:  [ 
  'Aragon',
  'ANT',
  "['0xb4124cEB3451635DAcedd11767f004d8a28c6eE7']",
  '1' 
]
```

We could do some parsing similar to [yargs/docs/tricks](https://github.com/yargs/yargs/blob/master/docs/tricks.md), i.e.: 
- if `JSON.parse(arg)` works then override with that

Thoughts?

**Comment by ottodevs** [comment]
Does it make sense to add a param for `dao new` like: `--config-file` to pass a json with the needed array params?

**Comment by 0x6431346e** [comment]
> Does it make sense to add a param for `dao new` like: `--config-file` to pass a json with the needed array params?
> Another solution that comes to my mind, could be parsing params and split them on commas maybe to get array elements on each one, but this one seems less "standard" for me than the config file

That should work! Perhaps calling it `--fn-args-file` instead?

At least as an interim solution before we implement proper parsing of the arguments.

**Comment by 0x6431346e** [comment]
Related to #555 and https://github.com/aragon/aragon-apps/issues/630

**Comment by 0x6431346e** [comment]
This will be fixed in #555 , but coercing based on the ABI would be much nicer so let's keep this open. :point_down: 	

> We could potentially do some coercing ourselves if we know the ABI of the kit function being called at that stage.
> 
> I think you should still be able to provide a single string (e.g. `'0xb4124...c6eE7'`) to that argument slot, but if it becomes more than one, then you should need to use the array syntax (e.g. `'["0xb4124...c6eE7"]'`). If the ABI tells us one of its arguments is an array, then do some parsing ([perhaps using a small json-schema snippet](https://github.com/epoberezkin/ajv#coercing-data-types)?) to produce an array value.
